### PR TITLE
Support Memory Browser for multiple child

### PR DESCRIPTION
**Pull request by trongle0504**
Support Memory Browser for multiple children

### CODE REVIEW

File: src/GDBDebugSession.ts [hunk 1]
```diff
@@ -198,7 +200,12 @@ export class GDBDebugSession extends LoggingDebugSession {
         args: any
     ): void {
         if (command === 'cdt-gdb-adapter/Memory') {
-            this.memoryRequest(response as MemoryResponse, args);
+            if (args.address === '') {
```
Comment (jonahgraham) [inline src/GDBDebugSession.ts:203]: Any changes in cdt-gdb-adapter require tests. Please provide a test along with explanation about what this code is changing.

**Review by jonahgraham — CHANGES_REQUESTED** [review]
I haven't looked at the next layers in the stack yet, but these changes don't look like they belong here. Especially a required new field in an arguments interface requires strong justification/explanation and tests.

File: src/GDBDebugSession.ts [hunk 2]
```diff
@@ -96,6 +97,7 @@ export interface MemoryContents {
     /* Hex-encoded string of bytes.  */
     data: string;
     address: string;
+    child?: string[];
```
[the remark this answers is not in the document]
Comment (trongle0504) [reply src/GDBDebugSession.ts]: It doesn't remain after changing the code.

File: src/GDBDebugSession.ts (hunk 1 above)
Comment (trongle0504) [reply src/GDBDebugSession.ts]: It doesn't remain after changing the code

File: src/GDBDebugSession.ts [hunk 3]
```diff
@@ -201,6 +202,10 @@ export class GDBDebugSession extends LoggingDebugSession {
             this.memoryRequest(response as MemoryResponse, args);
             // This custom request exists to allow tests in this repository to run arbitrary commands
             // Use at your own risk!
+        } else if ('cdt-amalgamator/getChildDapNames') {
```
Comment (jonahgraham) [inline src/GDBDebugSession.ts:205]: The adapter doesn't/shouldn't need to know about custom messages of the amalgamator. The client code can issue this command, and handle the case where it fails.

**Review by jonahgraham — CHANGES_REQUESTED** [review]
I think this change should simply be closed.

**Comment by trongle0504** [comment]
No need changes with new solution

**Pull request closed without merging**